src/components/SignupForm.jsx

```diff
@@ -41,45 +50,81 @@ const SubmitButton = styled.button`
   border-radius: 4px;
   cursor: pointer;
 `;
+const StyledLink = styled(Link)`
+  color: #007bff;
+  text-decoration: none;
+  margin-left: 10px;
+`;
 
 const SignupForm = () => {
-  const [username, setUsername] = useState('');
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
+  const [username, setUsername] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
   const [errors, setErrors] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
@ElisaTumwebaze please add this to the same hook and use that single hook other than creating these multiple hooks